### PR TITLE
[FIX] Bug #6975 Change is now sent to the right cash journal, not always the first one

### PR DESCRIPTION
**Pull request by alexis-via**
Please refer to my bug report #6975 for a step-by-step scenario to reproduce the bug on an up-to-date Odoo v8

### CODE REVIEW

File: addons/point_of_sale/point_of_sale.py [hunk 1]
```diff
@@ -590,16 +593,15 @@ def _process_order(self, cr, uid, order, context=None):
         if not float_is_zero(order['amount_return'], self.pool.get('decimal.precision').precision_get(cr, uid, 'Account')):
             cash_journal = session.cash_journal_id
             if not cash_journal:
-                cash_journal_ids = filter(lambda st: st.journal_id.type=='cash', session.statement_ids)
-                if not len(cash_journal_ids):
+                cash_journals = filter(lambda journal: journal.type=='cash', self.pool['account.journal'].browse(cr, uid, journal_ids, context=context))
```
Comment (mart-e) [inline addons/point_of_sale/point_of_sale.py]: could be replaced by a search `[('type', '=', 'cash'), ('id', 'in', journal_ids)]`

**Comment by mart-e** [comment]
Hello,

Your patch will not work if we are not using cash as payment method. Let say I pay by debit card more than needed and ask to receive cash back (I do it when I am too lazy to go to the ATM :) )
So I would say, prioritize on current method if it's cash, if not, use the first cash method available.

**Comment by alexis-via** [comment]
@mart-e OK, I didn't think about the scenario to over-pay by credit card and ask for cash back. I'll adapt my PR.

**Comment by alexis-via** [comment]
@mart-e I took into account your remark in my last commit and re-tested the regular scenario and the scenario you described, and it works well with the 2 scenarios.

**Comment by mart-e** [comment]
Thank you for the update, merged at 4b548d98d112b6d2d32b95d8d89624f0fac0426b

**Comment by gmoncada** [comment]
Hi, This fix don't work with "Cash Control" in Payment Methods.
pos_bug

**Comment by mart-e** [comment]
fixed at 0fb1b6a8568b599741eda2497a8d6b6339df2783

**Pull request closed without merging**